axelrod/strategies/adaptivezerodet.py

```diff
@@ -0,0 +1,120 @@
+from axelrod.action import Action
+from axelrod.player import Player
+from axelrod.random_ import random_choice
+from typing import Tuple
+from typing import List
+
+C, D = Action.C, Action.D
+
+class AdaptiveZeroDet(Player):
+	name = 'AdaptiveZeroDet'
+	classifier = {
+        'memory_depth': float('inf'),  # Long memory
+        'stochastic': True,
+        'makes_use_of': set(["game"]),
+        'long_run_time': False,
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+	def __init__(self, phi: float = 0.125, s: float = 0.5, l: float = 3, four_vector: Tuple[float, float, float, float] = None, initial: Action = C) -> None:
+		# This Keeps track of the parameter values (phi,s,l) as well as the four vector which makes final decisions.
+		self.scores = {C: 0, D: 0}
+		self.phi = phi
+		self.s = s
+		self.l = l
+		self._initial = initial
+		super().__init__()
+
+	def set_four_vector(self, four_vector: Tuple[float, float, float, float]):
+		# This checks the four vector is usable and allows previous matches' output to be input for next four vector
+		if not all(0 <= p <= 1 for p in four_vector):
+			raise ValueError("An element in the probability vector, {}, is not between 0 and 1.".format(str(four_vector)))
+		self._four_vector = dict(zip([(C, C), (C, D), (D, C), (D, D)], map(float, four_vector)))
+		self.classifier['stochastic'] = any(0 < x < 1 for x in set(four_vector))
+
+	def score_last_round(self, opponent: Player):
+		# This gives the strategy the game attributes and allows the strategy to score itself properly

```

Comment:
Could we change this to a docstring, ie use the triple:

```
"""
```

Similarly for our other ones.